packages/entity-server/src/@types/express/index.d.ts

```diff
@@ -3,20 +3,19 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 import { AccessPolicy } from '@tupaia/access-policy';
-import { Authenticator } from '@tupaia/auth';
-import { EntityServerModelRegistry, Context } from '../../types';
+import { Context } from '@tupaia/server-boilerplate';
+import { EntityServerModelRegistry } from '../../types';
 
 declare global {
   namespace Express {
     export interface Request {
       accessPolicy: AccessPolicy;
-      authenticator: Authenticator;
       models: EntityServerModelRegistry;
-      context: Context;
+      ctx: Context;
     }
 
     export interface Response {
-      context: Context;
+      ctx: Context;
```

Comment:
We need these `index.d.ts` declaration files to extend the fields available on the standard request object. Using standard `interface OurRequest extends Request { ourField: string};` runs in to a bunch of problems when trying to use express library functions.

I tried really hard to see if we could share a single `index.d.ts` between various packages but it just seemed impossible 😢 

... but fear not! While it does add a bit of extra boilerplate (the whole thing we've been trying to get away from hehe) it's actually proven to be really neat to be able to specify these in a per-package kind of way. For instance, the `models` field on requests is actually used by both the `server-boilerplate` package and the `entity-server` package. But these two packages are interested in different types of models. By being able to type cast this object per package, we can restrict the scope of the models object that each package is aware of to just the database tables that matter to them 👍

Reply:
Luckily it's not much boilerplate, since we can still load types for each field (eg `ModelRegistry` from `@tupaia/database` for `Request.models`)! So all good 😃